client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -378,6 +390,11 @@ static String endpoint(String[] indices, String[] types, String endpoint) {
         return endpoint(String.join(",", indices), String.join(",", types), endpoint);
     }
 
+
+    static String endpoint(String[] indices, String endpoint) {
```

Comment:
The problem is that it would then clash with the method below:

```
    /**
     * Utility method to build request's endpoint.
     */
    static String endpoint(String... parts) {
        StringJoiner joiner = new StringJoiner("/", "/", "");
        for (String part : parts) {
            if (Strings.hasLength(part)) {
                joiner.add(part);
            }
        }
        return joiner.toString();
    }
```

To be fair, having multiple `endpoint` methods that all take String collections as parameters, but have different behaviour, does feel like asking for trouble. How about renaming one or the other to `buildEndpoint`, `buildPath`, or similar?

Reply:
Alternatively, I could just not add a new method and call:

```
  String endpoint = endpoint(deleteIndexRequest.indices(), new String[0], "");
```

Would that be preferable?